### DISCUSSION TITLE
Gradle fetching library from oss.sontatype.org throws 403 although library is public

### DISCUSSION THREAD

**Discussion by Siedlerchr** [community/community#25201] [Actions]
<p>We have a java application that depends on two snapshot libraries from <a href="http://oss.sonatype.org" rel="nofollow noopener">oss.sonatype.org</a><br>
However,  we noticed that it throws always a 403 error when running on github actions.</p>
<p>Manually fetching that library on my system or accessing it in the browser works</p>
<p>Are the runners from github actions blacklisted?</p>
<pre><code>&gt; Could not resolve all files for configuration ':compileClasspath'.
   &gt; Could not resolve de.saxsys:mvvmfx-validation:1.9.0-SNAPSHOT.
     Required by:
         project :
      &gt; Could not resolve de.saxsys:mvvmfx-validation:1.9.0-SNAPSHOT.
         &gt; Unable to load Maven meta-data from https://oss.sonatype.org/content/groups/public/de/saxsys/mvvmfx-validation/1.9.0-SNAPSHOT/maven-metadata.xml.
            &gt; Could not HEAD 'https://oss.sonatype.org/content/groups/public/de/saxsys/mvvmfx-validation/1.9.0-SNAPSHOT/maven-metadata.xml'. Received status code 403 from server: Forbidden
   &gt; Could not resolve de.undercouch:citeproc-java:2.1.0-SNAPSHOT.
     Required by:
         project :
      &gt; Could not resolve de.undercouch:citeproc-java:2.1.0-SNAPSHOT.
         &gt; Unable to load Maven meta-data from https://oss.sonatype.org/content/groups/public/de/undercouch/citeproc-java/2.1.0-SNAPSHOT/maven-metadata.xml.
            &gt; Could not HEAD 'https://oss.sonatype.org/content/groups/public/de/undercouch/citeproc-java/2.1.0-SNAPSHOT/maven-metadata.xml'. Received status code 403 from server: Forbidden
</code></pre>

**Comment by erikc5000** [comment]
<p>I’m seeing the exact same issue as of a few days ago. I can copy-paste the URL into my web browser and access maven-metadata.xml, but apparently the GitHub Actions runner can’t.</p>

**Comment by stefan-kolb** [comment]
<p>You should probably add that this only affects macOS runners, as far as I can see?!</p>

**Comment by Siedlerchr** [comment]
<p>I think its due to caching, we had not enabled gradle caching on Mac because the limit was exceeded.<br>
But on Linux and windows we have gradle caching, so I guess that’s because it fails not there.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/siedlerchr">@siedlerchr</a>,</p>
<p>I confirm the two snapshot libraries can be accessed from my side, could you please try again?</p>
<p>If it persists, could you please share more details about the yaml file/action? </p>
<p>Thanks.</p>

**Comment by Siedlerchr** [comment]
<p>I’m still getting the rror:</p>
<p><a href="https://github.com/JabRef/jabref/runs/510500180" rel="nofollow noopener">https://github.com/JabRef/jabref/runs/510500180</a></p>
<p>Here is the relevant yaml part:</p>
<p><a href="https://github.com/JabRef/jabref/blob/cbd26f11f54f4a46c2597b6ba564b76aa9d68c88/.github/workflows/deployment.yml#L97-L103" rel="nofollow noopener">https://github.com/JabRef/jabref/blob/cbd26f11f54f4a46c2597b6ba564b76aa9d68c88/.github/workflows/deployment.yml#L97-L103</a></p>
<p>Gradle contains:</p>
<p>implementation ‘de.undercouch:citeproc-java:2.1.0-SNAPSHOT’<br>
implementation ‘de.saxsys:mvvmfx-validation:1.9.0-SNAPSHOT’</p>

**Comment by frankfliu** [comment]
<p>I’m also seeing the same issue since 3/12/2019, everything was working fine before that.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/siedlerchr">@siedlerchr</a> ,</p>
<p>Thanks for your reply! Confirmed with Maven center, the macOS hosted runner ip is blocked, and they has re-enabled this ip.</p>
<p>some info for you:</p>
<p>“We need to ask the customer to enable Gradle caching on their workflow, or we will hit this issue again. This also blocks all other users on the same mac pool when the public IP is blocked.”</p>
<p>I forked your repo, and re-run the workflow, it’s passed on all platform including macOS now! Please check your workflow for confirmation, thanks.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7cb1db9ffe44bf9a3da103282eea39d6b44c44bb.png" data-download-href="/uploads/short-url/hN6greNzmjzpx2R9654C840LAh5.png?dl=1" title="macOSMaven.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181089371-678ffc5c-5d2b-42a9-8904-d9a826847410.png" alt="macOSMaven.png" data-base62-sha1="hN6greNzmjzpx2R9654C840LAh5" width="473" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7cb1db9ffe44bf9a3da103282eea39d6b44c44bb_2_473x500.png, https://aws1.discourse-cdn.com/github/original/2X/7/7cb1db9ffe44bf9a3da103282eea39d6b44c44bb.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/7/7cb1db9ffe44bf9a3da103282eea39d6b44c44bb.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7cb1db9ffe44bf9a3da103282eea39d6b44c44bb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">macOSMaven.png</span><span class="informations">528×557 65.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Siedlerchr** [comment]
<p>Thanks, it works now!</p>
<p>&gt;“We need to ask the customer to enable Gradle caching on their workflow, or we will hit this issue again. This also blocks all other users on the same mac pool when the public IP is blocked.”</p>
<p>We have caching now enabled on mac os as well. We had it disabled, because previously the cache was over the initial 2GB limit. Now as the cache limit got increased to 5.0 it’s working again</p>
